housing_cooperative_base/wizard/rental_state_year_report_wizard.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, fields, models
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class RentalStateYearReportWizard(models.TransientModel):
+    _name = "hc.rental.state.year.report.wizard"
+    _description = "Create Rental State Year Report"
+
+    name = fields.Char(compute="_compute_name", store=True)
+    year = fields.Integer(
+        string="Year",
+        required=True,
+        default=fields.Date.today().strftime("%Y"),
+    )
+    building_ids = fields.Many2many(
+        comodel_name="hc.building",
+        string="Buildings",
+        required=True,
+        default=lambda s: s.env["hc.building"].search([]),
+    )
+    lease_line_ids = fields.Many2many(
+        string="Selected Leases",
+        comodel_name="hc.lease.line",
+        compute="_compute_lease_line_ids",
+    )
+    premise_ids = fields.Many2many(
+        string="Selected Premises",
+        comodel_name="hc.premise",
+        compute="_compute_premise_ids",
+    )
+
+    @api.multi
+    @api.depends("year")
+    def _compute_name(self):
+        for wizard in self:
+            wizard.name = "rental_state_year_report_%s" % wizard.year
+
+    @api.multi
+    @api.depends("year", "building_ids")
+    def _compute_lease_line_ids(self):
+        if not self.year:
+            self.year = fields.Date.today().strftime("%Y")
+        self.ensure_one()
+        self.lease_line_ids = (
+            self.env["hc.lease.line"]
+            .search([])
+            .filtered(
+                lambda lease_line_id: int(lease_line_id.start.strftime("%Y"))
+                <= int(self.year)
+                <= int(lease_line_id.end.strftime("%Y"))
+            )
+            .filtered(
+                lambda lease_line_id: lease_line_id.building_id
+                in self.building_ids
+            )
+        )
+        return True
+
+    @api.multi
+    @api.depends("year", "building_ids")
+    def _compute_premise_ids(self):
+        self.ensure_one()
+        self.premise_ids = (
+            self.env["hc.premise"]
+            .search([])
+            .filtered(
```

Comment:
Same remark.

Reply:
Can I do the year filter in search as well? I'm asking since it applies a function on the field.

Is this allowed: `("start.year", "<=", self.year)`, if only `start` is a field?